### ISSUE TITLE
Inclusive `par_for` for raw arrays (physics kernels)

### ISSUE THREAD

**Issue by Montana** [parthenon-hpc-lab/parthenon#733]
Hey all, 

I seen a topic on this earlier - was  `par_for` with exclusive end index ever implemented? One of the suggestions was to name it `par_efor` or something similar. Most of the AMR and physics kernels that the current inclusive `par_for` makes sense, but occasionally when dealing with raw arrays an exclusive `for` would be more convenient in my opinion.

**Comment by Yurlungur** [comment]
No---we never implemented this, thought it would be straightforward to do so.

**Comment by Montana** [comment]
Thank you @Yurlungur.
